### ISSUE TITLE
overloading __getattr__()

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#2739]
### Discussed in https://github.com/microsoft/pyright/discussions/2737

<div type='discussions-op-text'>

<sup>Originally posted by **lovetox** December 18, 2021</sup>
Hi, is it possible to overload __getattr__()

```python3
from typing import Any
from typing import overload
from typing import Literal

class Obj:
    @overload
    def __getattr__(self, name: Literal["foo"]) -> int: ...
    @overload
    def __getattr__(self, name: Literal["bar"]) -> str: ...
    def __getattr__(self, name: str) -> Any:
        if name == 'foo':
            return 1
        return '1'

a = Obj()
b = a.foo
c = a.bar

b = getattr(a, 'foo')
c = getattr(a, 'bar')
```

so obviously foo and bar is not recognized by pyright as Literal, but i have no idea what i could use otherwise.</div>

**Comment by erictraut** [comment]
This will be included in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.198, which I just published. It will also be included in the next release of pylance.
